runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java

```diff
@@ -55,7 +57,18 @@ public static TestFlinkRunner create(boolean streaming) {
   @Override
   public PipelineResult run(Pipeline pipeline) {
     try {
-      return delegate.run(pipeline);
+      PipelineResult result = delegate.run(pipeline);
+      // For tests that don't call waitUntilFinish().
+      try {
+        result.waitUntilFinish(Duration.standardSeconds(30));
```

Comment:
Why this duration?

Also the nested tries and catches and exception propagation make my head hurt a little. I'm not sure off the top of my head if there's any other way to do it.

You should also consider only calling `cancel` if the job has not already terminated.

Also it turns out the FlinkRunner doesn't support `cancel`